src/components/MyAccount/NewSettings/EmailForm.test.tsx

```diff
@@ -0,0 +1,138 @@
+import { render, screen, fireEvent, waitFor } from "@testing-library/react"
+import EmailForm from "./EmailForm"
+import { filteredPickupLocations } from "../../../../__test__/fixtures/processedMyAccountData"
+import { PatronDataProvider } from "../../../context/PatronDataContext"
+import { processedPatron } from "../../../../__test__/fixtures/processedMyAccountData"
+
+describe("email form", () => {
+  const mockSetIsSuccess = jest.fn()
+  const mockSetIsFailure = jest.fn()
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+    global.fetch = jest.fn().mockResolvedValue({
+      json: async () => {
+        console.log("Updated")
+      },
+      status: 200,
+    } as Response)
+  })
+
+  const component = (
+    <PatronDataProvider
+      value={{
+        patron: processedPatron,
+        pickupLocations: filteredPickupLocations,
+      }}
+    >
+      <EmailForm
+        patronData={processedPatron}
+        setIsSuccess={mockSetIsSuccess}
+        setIsFailure={mockSetIsFailure}
+      />
+    </PatronDataProvider>
+  )
+
+  it("renders correctly with initial email", () => {
+    render(component)
+
+    expect(screen.getByText("streganonna@gmail.com")).toBeInTheDocument()
+
+    expect(screen.getByRole("button", { name: /edit/i })).toBeInTheDocument()
```

Comment:
Is there a reason you're using a regex here? Not concerned, just curious.

Reply:
No particular reason I guess, would it be better to be all regex or all strings?

Reply:
I _think_ strings are suppose to be a faster lookup, but it might be negligible. The bottleneck, if any, might be the `getByRole` function, but either regex or string is ok.